src/fairseq2/recipes/eval/asr.py

```diff
@@ -74,148 +75,194 @@ class AsrEvalConfig(HFEvalConfig):
-def postprocesser(
-    outputs: Any, targets: SequenceBatch, tokenizer: TextTokenizer
-) -> Tuple[List[str], List[str]]:
-    decoder = tokenizer.create_decoder()
-    pad_idx = tokenizer.vocab_info.pad_idx
-
-    hypotheses, _ = outputs.generate_hypotheses(pad_idx=pad_idx)
-    predictions = [decoder(item) for item in hypotheses]
-    references = [decoder(item) for item in targets.seqs.to(torch.int32)]
-
-    return predictions, references
-
-
-@hf_presets.decorator("librispeech_asr")
-def _librispeech_asr_config() -> AsrEvalConfig:
+@wav2vec2_presets.decorator("librispeech_asr")
+def _wav2vec2_librispeech_asr_config() -> AsrEvalConfig:
     return AsrEvalConfig(
         dataset_name="librispeech_asr",
         model_name="wav2vec2_asr_base_10h",
-        split="test.other"
-        # converter=librispeech_asr_to_batch,
+        split="test.other",
     )
 
 
-def load_wav2vec2_asr_evaluator(
-    config: HFEvalConfig, output_dir: Path
-) -> HFEvaluator[Seq2SeqBatch]:
-    """
-    Load the evaluator used for downstream evaluation of the model
-    in a downstream dataset and report BLEU scores
-
-    Args:
-        config (HFEvalConfig): The configuration for the evaluation.
-        output_dir (Path): The output directory to store the evaluation results.
-
-    Returns:
-        HFEvaluator: Evaluation process results.
-    """
-    if not isinstance(config, AsrEvalConfig):
-        raise ValueError(f"Expect AsrEvalConfig, get {type(config)}")
-
-    iterable_ds = load_dataset(config.dataset_name, split=config.split, streaming=True)
-    max_samples = config.max_samples if config.max_samples is not None else math.inf
-    # Load a subset of the dataset if max_samples is set
-    ds = Dataset.from_generator(
-        lambda: (
-            yield from (
-                item for idx, item in enumerate(iterable_ds) if idx < max_samples
-            )
-        ),
-        features=iterable_ds.features,
-    )
-
-    gang = setup_root_gang(log)
-
-    if gang.rank == 0:
-        init_device = gang.device
-    else:
-        init_device = META
-
-    tokenizer = load_text_tokenizer(config.tokenizer_name)
-    encoder = tokenizer.create_encoder(device=init_device)
-
-    ds = ds.map(lambda x: _preprocess_example(x, encoder, init_device))
-    format = {
-        "type": "torch",
-        "format_kwargs": {"dtype": torch.float16, "device": init_device},
-    }
-    ds.set_format(**format, columns=["audio", "text"])
-
-    pipeline_reader = create_hf_reader(
-        dataset=ds,
-        gang=gang,
-        converter=_librispeech_asr_to_batch,
-        batching=StaticBatching(config.max_num_elements),
-        num_prefetch=config.num_prefetch,
-        pad_value=tokenizer.vocab_info.pad_idx,
-        max_seq_len=config.max_audio_len,
-    )
-
-    model = load_wav2vec2_asr_model(
-        config.model_name, device=init_device, dtype=config.dtype
+@whisper_presets.decorator("librispeech_asr")
+def _whisper_librispeech_asr_config() -> AsrEvalConfig:
```

Comment:
One default preset is enough. If the user wants Whisper with librispeech_asr or other datasets, then they should specify it directly at runtime.

Otherwise we will have MxN presets for each of M models and N datasets :).